ocs_ci/ocs/tests/test_fiojob.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf8 -*-
+
+from unittest.mock import patch, Mock
+import logging
+import os
+import textwrap
+
+import pytest
+import yaml
+import yaml.parser
+
+from ocs_ci.ocs import fiojob
+
+
+HERE = os.path.abspath(os.path.dirname(__file__))
+
+
+@pytest.fixture
+def fio_json_output():
+    """
+    Example of fio --output-format=json output. Based on actual test run.
+    """
+    filename = os.path.join(HERE, "fio.output")
+    with open(filename, 'r') as fio_output:
+        content = fio_output.read()
+    return content
+
+
+@pytest.fixture
+def fio_json_output_with_error(fio_json_output):
+    """
+    Example of fio --output-format=json output, with io_u error. Based on
+    actual test run.
+    """
+    err_line = (
+        "fio: io_u error on file /mnt/target/simple-write.0.0: "
+        "No space left on device: write offset=90280222720, buflen=4096")
+    return err_line + "\n" + fio_json_output
+
+
+@pytest.fixture
+def fio_json_output_broken():
+    """
+    Example of broken, unparseable json output.
+    """
+    fio_out = textwrap.dedent("""
+        {
+          "fio version" : "fio-3.7",
+          "timestamp" : 1584531581,
+          "timestamp_ms" : 1584531581691,
+          "time" : "Wed Mar 18 11:39:41 2020",
+          "jobs" : [
+        """)
+    return fio_out
+
+
+def test_fio_to_dict_empty():
+    assert fiojob.fio_to_dict("") is None
+
+
+def test_fio_to_dict_without_error(fio_json_output):
+    fio_dict = fiojob.fio_to_dict(fio_json_output)
+    assert type(fio_dict) == dict
+    assert len(fio_dict['jobs']) == 1
+
+
+def test_fio_to_dict_with_error(fio_json_output_with_error):
+    fio_dict = fiojob.fio_to_dict(fio_json_output_with_error)
+    assert type(fio_dict) == dict
+    assert len(fio_dict['jobs']) == 1
+
+
+def test_fio_to_dict_output_broken(fio_json_output_broken, caplog):
+    caplog.set_level(logging.ERROR)
+    with pytest.raises(yaml.parser.ParserError):
+        fiojob.fio_to_dict(fio_json_output_broken)
+    assert "json output from fio can't be parsed" in caplog.text
+
+
+@pytest.fixture
+def ceph_df_json_output_clusterempty():
+    """
+    Example of output from ``ceph df --format json-pretty`` command.
+    Based on actual test run:
+    jnk-ai3c33-t1/jnk-ai3c33-t1_20200318T095635/logs/ocs-ci-logs-1584528704
+    """
+    ceph_df_out = textwrap.dedent("""
```

Comment:
same comment as in my last PR, why not move this in template and dynamically load, the consistency we have across scripts makes it easier to read.